File: day_3/Valid_palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
let isPalindrome = function (s) {
    let re = /[a-zA-Z0-9]+/gi,
        complete_str = [...s.matchAll(re)].join("").toLowerCase(),
        word_len = complete_str.length;
    for (let i = 0; i <= word_len/2; i++){
        if (complete_str[i] !== complete_str[word_len-i-1]) return false;
    }
    return true;
};

File: day_3/Valid_Palindrome.py
import re


def isPalindrome(s: str) -> bool:
    s = "".join(re.findall(r"[A-Za-z0-9]+", s))
    left = 0
    right = len(s) - 1

    while left <= right:
        if s[left].lower() != s[right].lower():
            return False
        left += 1
        right -= 1
    return True

